great_expectations/validator/validation_graph.py

```diff
@@ -28,20 +40,190 @@ def id(self):
+                    logger.warning(
+                        f"Metric {str(metric_configuration.id)} has created a circular dependency"
+                    )
+                    continue
+                self.add(
+                    MetricEdge(
+                        left=metric_configuration,
+                        right=metric_dependency,
+                    )
+                )
+                self.build_metric_dependency_graph(
+                    metric_configuration=metric_dependency,
+                    runtime_configuration=runtime_configuration,
+                )
+
+    def resolve_validation_graph(  # noqa: C901 - complexity 16
+        self,
+        metrics: Dict[Tuple[str, str, str], Any],
+        runtime_configuration: Optional[dict] = None,
+        min_graph_edges_pbar_enable: int = 0,  # Set to low number (e.g., 3) to suppress progress bar for small graphs.
+        show_progress_bars: bool = True,
+    ) -> Dict[
+        Tuple[str, str, str],
+        Dict[str, Union[MetricConfiguration, Set[ExceptionInfo], int]],
+    ]:
+        if runtime_configuration is None:
+            runtime_configuration = {}
+
+        if runtime_configuration.get("catch_exceptions", True):
+            catch_exceptions = True
+        else:
+            catch_exceptions = False
+
+        failed_metric_info: Dict[
+            Tuple[str, str, str],
+            Dict[str, Union[MetricConfiguration, Set[ExceptionInfo], int]],
+        ] = {}
+        aborted_metrics_info: Dict[
+            Tuple[str, str, str],
+            Dict[str, Union[MetricConfiguration, Set[ExceptionInfo], int]],
+        ] = {}
+
+        ready_metrics: Set[MetricConfiguration]
+        needed_metrics: Set[MetricConfiguration]
+
+        exception_info: ExceptionInfo
+
+        progress_bar: Optional[tqdm] = None
+
+        done: bool = False
+        while not done:
+            ready_metrics, needed_metrics = self._parse(metrics=metrics)
+
+            # Check to see if the user has disabled progress bars
+            disable = not show_progress_bars
+            if len(self.edges) < min_graph_edges_pbar_enable:
+                disable = True
+
+            if progress_bar is None:
+                # noinspection PyProtectedMember,SpellCheckingInspection
+                progress_bar = tqdm(
+                    total=len(ready_metrics) + len(needed_metrics),
+                    desc="Calculating Metrics",
+                    disable=disable,
+                )
+            progress_bar.update(0)
+            progress_bar.refresh()
+
+            computable_metrics = set()
+
+            for metric in ready_metrics:
+                if metric.id in failed_metric_info and failed_metric_info[metric.id]["num_failures"] >= MAX_METRIC_COMPUTATION_RETRIES:  # type: ignore[operator]  # Incorrect flagging of 'Unsupported operand types for <= ("int" and "MetricConfiguration") and for >= ("Set[ExceptionInfo]" and "int")' in deep "Union" structure.
```

Comment:
I think this is fixable with some updates to the `MetricConfiguration` operator typing
But I think it's fine to defer this to another PR

Reply:
@Kilo59 Thanks -- we can address in a follow-on PR.